parley/src/resolve/mod.rs

```diff
@@ -486,7 +489,17 @@ impl<B: Brush> ResolvedStyle<B> {
             StrikethroughOffset(value) => self.strikethrough.offset == *value,
             StrikethroughSize(value) => self.strikethrough.size == *value,
             StrikethroughBrush(value) => self.strikethrough.brush == *value,
-            LineHeight(value) => nearly_eq(self.line_height, *value),
+            LineHeight(value) => match (self.line_height, *value) {
+                (crate::LineHeight::MetricsRelative(a), crate::LineHeight::MetricsRelative(b))
+                | (
+                    crate::LineHeight::FontSizeRelative(a),
+                    crate::LineHeight::FontSizeRelative(b),
+                )
+                | (crate::LineHeight::Absolute(a), crate::LineHeight::Absolute(b)) => {
+                    nearly_eq(a, b)
+                }
+                _ => false,
```

Comment:
A couple of notes on this one:
- Consider importing `crate::LineHeight` at the top of the file (partly for consistency with surrounding code, partly because I think the `match` would format better like that)
- Consider making this a method on `LineHeight`.